main.tf

```diff
@@ -2165,6 +2165,16 @@ locals {
       scope       = "parent"
       regex       = "^[a-zA-Z0-9-_]{1,128}$"
     }
+    nat_gateway = {
+      name        = substr(join("-", compact([local.prefix, "ng", local.suffix])), 0, 80)
+      name_unique = substr(join("-", compact([local.prefix, "ng", local.suffix_unique])), 0, 80)
+      dashes      = true
+      slug        = "ng"
+      min_length  = 1
+      max_length  = 80
+      scope       = "resourceGroup"
+      regex       = "^(?=.{1,80}$)[a-zA-Z0-9][a-zA-Z0-9-._]+[a-zA-Z0-9_]$"
```

Comment:
regex is faulty .. it should be like

  regex: '^[a-zA-Z0-9][a-zA-Z0-9]([a-zA-Z0-9]|-){0,77}[a-zA-Z0-9]$'